File: Books/Math_Refresher/python/central_limit_theorem.py
# central_limit_theorem.py
import numpy as np
import matplotlib.pyplot as plt
from coin_flip import t  # Import the coin_flip array

# Close any open figures
plt.close('all')

# Calculate the distribution
n = 1000
# (Alternatively, you could generate random numbers: t = np.random.rand(n))
trials = np.arange(1, n+1)
heads = 0
P_heads = np.ones_like(t, dtype=float)  # initialize array with the same shape as t

for ii in range(n):
    if t[ii] < 0.5:
        heads += 1
    P_heads[ii] = heads / (ii + 1)  # note: Python indexing starts at 0

# Animate plot of trial number vs. probability of heads
plt.figure(1)
for ii in range(n):
    plt.clf()  # Clear the current figure
    plt.plot(trials[:ii+1], P_heads[:ii+1], linewidth=2)
    plt.xlim([-50, 1000])
    plt.ylim([0, 1])
    plt.xlabel('Trial Number')
    plt.ylabel('Probability of Heads after n trials')
    plt.title('Trial Number vs Percent Heads')
    plt.pause(0.001)  # Pause briefly to update the plot

# Save the final figure
plt.savefig("../figures/law-of-large-numbers.pdf")
plt.show()

File: Books/Math_Refresher/matlab/central_limit_theorem.m
% Central Limit Theorem

% Start of script
%-------------------------------------------------------------------------%
close all;                   	% close all figures
clearvars; clearvars -global;	% clear all variables
clc;                         	% clear the command terminal
format shortG;                 	% pick the most compact numeric display
format compact;                	% suppress excess blank lines

% load coin flip
coin_flip

% calculate distribution
n = 1000;
% t = rand(n,1);
trials = 1:n;
heads = 0;
P_heads = ones(size(t));
for ii = 1:n
    if (t(ii) < 0.5)
        heads = heads + 1;
    end
    P_heads(ii) = heads/ii;  
end

% animate plot
figure(1)
for ii = 1:n
    plot(trials(1:ii),P_heads(1:ii),'LineWidth',2)
    xlim([-50 1000])
    ylim([0 1]);
    xlabel('Trial Number')
    ylabel('Probabilty of Heads after n trials')
    title ('Trial Number vs Percent Heads')
    drawnow;
end

save_all_figs_OPTION('../figures/law-of-large-numbers','pdf')
